utils/DeployedContractsHelpers.ts

```diff
@@ -75,52 +121,82 @@ export async function getTreasury(hre: HardhatRuntimeEnvironment): Promise<Signe
 }
 
-export async function getBDStableBdxStakingRewards(hre: HardhatRuntimeEnvironment, stableSymbol: string) {
-  return await getBDStableCollateralStakingRewards(hre, stableSymbol, "BDX");
+export async function getStakingRewardsWithBdx(hre: HardhatRuntimeEnvironment, symbol: string): Promise<StakingRewards | undefined> {
+  return await getBDStableCollateralStakingRewards(hre, symbol, "BDX");
 }
 
-async function getBDStableCollateralStakingRewards(hre: HardhatRuntimeEnvironment, stableSymbol: string, collateralSymbol: string) {
-  const deployer = await getDeployer(hre);
-
+async function getBDStableCollateralStakingRewards(
+  hre: HardhatRuntimeEnvironment,
+  stableSymbol: string,
+  collateralSymbol: string
+): Promise<StakingRewards | undefined> {
   const collateralAddress = await getContratAddress(hre, collateralSymbol);
   const stableAddress = await getContratAddress(hre, stableSymbol);
   const poolKey = getPoolKey(collateralAddress, stableAddress, collateralSymbol, stableSymbol);
 
-  return (await hre.ethers.getContract(`StakingRewards_${poolKey}`, deployer)) as StakingRewards;
+  const stakingRewardsContract = await getContractOrUndefined(hre, `StakingRewards_${poolKey}`);
+  return stakingRewardsContract && (stakingRewardsContract as StakingRewards);
 }
 
-export async function getAllBDStableStakingRewards(hre: HardhatRuntimeEnvironment) {
+async function getContractOrUndefined(hre: HardhatRuntimeEnvironment, contractName: string): Promise<Contract | undefined> {
+  try {
+    const deployer = await getDeployer(hre);
+    return hre.ethers.getContract(contractName, deployer);
+  } catch {
+    return undefined;
+  }
+}
+
+export async function getAllBDStableStakingRewards(hre: HardhatRuntimeEnvironment): Promise<StakingRewards[]> {
   const deployer = await getDeployer(hre);
   const bdstablesSymbols = getAllBDStablesSymbols();
   const bdx = await getBdx(hre);
-  const stakingRewards = [];
+  const stakingRewards: StakingRewards[] = [];
   const stakingRewardsBdStablesMap = new Set<string>();
 
-  stakingRewards.push(await getBDStableWethStakingRewards(hre, await bdx.symbol()));
-  stakingRewards.push(await getBDStableWbtcStakingRewards(hre, await bdx.symbol()));
+  const bdxWethStakingRewards = await getStakingRewardsWithWeth(hre, await bdx.symbol());
+  if (bdxWethStakingRewards) {
+    stakingRewards.push(bdxWethStakingRewards);
+  }
+
+  const bdxWbtcStakingRewards = await getStakingRewardsWithWbtc(hre, await bdx.symbol());
+  if (bdxWbtcStakingRewards) {
+    stakingRewards.push(bdxWbtcStakingRewards);
+  }
 
   for (const symbolA of bdstablesSymbols) {
-    stakingRewards.push(await getBDStableWethStakingRewards(hre, symbolA));
-    stakingRewards.push(await getBDStableWbtcStakingRewards(hre, symbolA));
-    stakingRewards.push(await getBDStableBdxStakingRewards(hre, symbolA));
+    const wethStakingRewards = await getStakingRewardsWithWeth(hre, symbolA);
+    if (wethStakingRewards) {
```

Comment:
we probably should let it throw here, every stable is supposed to have a staking pair with the native token, right?

Reply:
I fixed the comments about the pairs with BDX since it won't change.
but why would we want to require having a pair with the native token?